docs/examples/guide/actions/suspend.py

```diff
@@ -0,0 +1,27 @@
+from code import InteractiveConsole
+import readline  # enable readline support in the REPL https://docs.python.org/3/library/readline.html
+
+from textual.app import App, ComposeResult
+from textual.widgets import Footer, Static
+
+
+class SuspendApp(App):
+    BINDINGS = [
+        ("r", "open_repl", "Open REPL"),
+    ]
+
+    def compose(self) -> ComposeResult:
+        yield Static(
+            "Press r to open the Python built-in REPL. Run quit() to return to the app."
+        )
+        yield Footer()
+
+    def action_open_repl(self):
+        with self.suspend():
+            repl = InteractiveConsole()
+            repl.interact()
```

Comment:
In #1150 I used
```
    def action_suspend(self) -> None:
        with self.suspend():
            print("Hi!")
            print("Resuming soon...")
            time.sleep(2)
```
as a reproduction case, which would simplify this example if desired, but I figured doing something interactive would be more interesting and inspiring.